source/application/REIXS/REIXSAppController.cpp

```diff
@@ -78,85 +78,134 @@ REIXSAppController::REIXSAppController(QObject *parent) :
 }
 
 bool REIXSAppController::startup()
-{
-	bool success = true;
+{	
+	if(AMAppController::startup()) {
 
-	success &= AMAppController::startup();
-	success &= AMActionRegistry3::s()->registerInfoAndAction<AMSamplePlatePre2013MoveActionInfo, AMSamplePlatePre2013MoveAction>("Move Sample Position", "Move to a different marked sample position", ":system-run.png");
-	success &= AMActionRegistry3::s()->registerInfoAndEditor<AMSamplePlatePre2013MoveActionInfo, AMSamplePlatePre2013MoveActionEditor>();
+		// Initialize the central beamline object
+		REIXSBeamline::bl();
+		// Initialize the storage ring.
+		CLSStorageRing::sr1();
 
-	return success;
-}
 
-// Re-implemented to create the REIXSBeamline object
-bool REIXSAppController::startupBeforeAnything() {
-	if(!AMAppController::startupBeforeAnything()) return false;
+		registerClasses();
+
+		// Checking for and making the first run in the database, if there isn't one already.
+		////////////////////////////////////////
+		AMRun existingRun;
+		if(!existingRun.loadFromDb(AMDatabase::database("user"), 1)) {
+			// no run yet... let's create one.
+			AMRun firstRun(CLSFacilityID::beamlineName(CLSFacilityID::REIXSBeamline), CLSFacilityID::REIXSBeamline); //5: REIXS Beamline
+			firstRun.storeToDb(AMDatabase::database("user"));
+		}
 
-	if (!AMChooseDataFolderDialog::getDataFolder("/AcquamanLocalData/reixs", "/home/reixs", "users"))
+		setupUserInterface();
+		makeConnections();
+	} else {
 		return false;
+	}
+}
 
-	// Initialize the central beamline object
-	REIXSBeamline::bl();
-	// Initialize the storage ring.
-	CLSStorageRing::sr1();
+void REIXSAppController::shutdown() {
 
-	connect(REIXSBeamline::bl()->spectrometer(), SIGNAL(connected(bool)), REIXSBeamline::bl()->spectrometer(), SLOT(updateGrating()));
+	// Make sure we release/clean-up the beamline interface
+	AMBeamline::releaseBl();
```

Comment:
Also, if this is the case across the board it's probably something that could be added to AMAppController::shutdown() instead of individually to each subclass.